src/test/kotlin/org/rust/ide/inspections/lints/RsUnusedLabelsInspectionTest.kt

```diff
@@ -0,0 +1,136 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.lints
+
+import org.rust.ide.inspections.RsInspectionsTestBase
+
+class RsUnusedLabelsInspectionTest : RsInspectionsTestBase(RsUnusedLabelsInspection::class) {
+    fun `test unused label with empty block`() = checkByText("""
+        fn main() {
+            /*warning descr="Unused label"*/'label:/*warning**/ {}
+        }
+    """)
+
+    fun `test unused label with while`() = checkByText("""
+        fn main() {
+            /*warning descr="Unused label"*/'label:/*warning**/ while 0 == 0 {}
+        }
+    """)
+
+    fun `test unused label with while let`() = checkByText("""
+        fn main() {
+            enum E<T> { A(T), B }
+            let opt = E::A(0);
+            /*warning descr="Unused label"*/'label:/*warning**/ while let E::A(_) = opt {}
+        }
+    """)
+
+    fun `test unused label with for`() = checkByText("""
+        fn main() {
+            /*warning descr="Unused label"*/'label:/*warning**/ for _ in 0..10 {}
+        }
+    """)
+
+    fun `test unused label with loop`() = checkByText("""
+        fn main() {
+            /*warning descr="Unused label"*/'label:/*warning**/ loop {}
+        }
+    """)
+
+    fun `test no unused label with nested for`() = checkByText("""
+        fn main() {
+            'a: for _ in 0..10 {
+                'b: for _ in 0..10 {
+                    break 'b;
+                }
+                break 'a;
+            }
+        }
+    """, ignoreExtraHighlighting = false)
```

Comment:
`ignoreExtraHighlighting = false` is the default value, so we can omit it here and at line 80